pyannote/database/custom.py

```diff
@@ -125,6 +130,9 @@ def subset_iter(database_name, file_lst=None, file_rttm=None, file_uem=None):
         # add 'annotated' if/when available
         if uri in annotated:
             current_file['annotated'] = annotated[uri]
+        elif uri in annotations:
+            # determine the duration of the file directly from the annotations
```

Comment:
"annotated" is usually larger than "annotation" extent (non-speech regions before/after the first/last speech turn). This is actually the case in DIHARD2 database (which was the initial reason for this pull request).

We should support providing both a list of RTTM/MDTM files and the corresponding UEM (which is the case for DIHARD). Not sure how... 
We would have to be smart in inferring the list of "uris" from all those (possibly disagreeing) files.

Reply:
I added this line because I noticed that several db plugins implemented it this way when they have no UEM files. You can see an example here: https://github.com/yinruiqing/pyannote-db-callhome/blob/81db9c8cc611322d10e9bddb3af0d72707124ec8/CallHome/__init__.py#L82

Reply:
This is in fact incorrect. 

`pyannote.database.get_annotated` is meant for this, and tries to be a bit smarter:
- whenever "audio" key is available, it will try to use the whole "audio" extent
- if "audio" is not available, it falls back to "annotation" extent

https://github.com/pyannote/pyannote-database/blob/ab00956315cb8f748665aefdec1eaddb354df67d/pyannote/database/util.py#L353-L391